caching.md

```diff
@@ -0,0 +1,53 @@
+## Introduction
+
+Intake provides easy access data sources from remote / cloud storage, however for large files the 
+cost of downloading files everytime data is read can be extremely high. We have developed a caching 
+strategy to store and manage data sources on the local file system.
+
+## How it works
+
+Intake catalogs are composed of data sources, which contain attributes descibing how to 
+load data files. To enable caching, catalog authors can now specify caching parameters 
+that will be used to download data file(s), store them in a configured local directory, 
+and return a their location on local disk. For example:
+
+```
+nyc_taxi:
+    description: NYC Taxi dataset
+    driver: csv
+    cache:
+      - argkey: urlpath
+        regex: 'datashader-data'
+        type: file
+    args:
+      urlpath: 's3://datashader-data/nyc_taxi.csv'
+      storage_options: {'anon': True}
+```
+
+Glob strings can be used for the ``urlpath``, which will be expanded to the individual files 
+listed in the base directory of the path. Each file will be downloaded in parallel and referrenced
+in the metadata.
+
```

Comment:
From the users' point of view, this happens transparently, they need know nothing about the caching policy, but they will notice that read times after the initial access are much faster.